packages/headless/src/controllers/smart-snippet-suggestions/headless-smart-snippet-suggestions.ts

```diff
@@ -0,0 +1,95 @@
+import {Engine} from '../../app/headless-engine';
+import {buildController, Controller} from '../controller/headless-controller';
+import {search} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {SearchSection} from '../../state/state-sections';
+
+export interface SmartSnippetSuggestionsDocumentId {
+  contentIdKey: string;
+  contentItValue: string;
+}
+
+/**
+ * The `SmartSnippetSuggestions` controller allows to manage additional queries for which a SmartSnippet model can provide relevant excerpts.
+ */
+export interface SmartSnippetSuggestions extends Controller {
```

Comment:
I prefer using the singular for controller names, because they return one instance, not an array.
I also feel the word `snippet` refers to the main answer, and does not need to be mentioned here.

Some thoughts:
- `SmartQuestionsList` (on the fence as to the word `smart`).
- `RelatedQuestionsList`
- `SimilarQuestionsList`
- `RelatedQuestionsManager`

I am leaning towards  `RelatedQuestionsList` personally.
Summoning @dlutzCoveo and @fbeaudoincoveo for additional ideas,

Reply:
* I am fairly certain "SmartSnippet" is part of the official marketing name... I'll ask in the appropriate channel to make sure. If that's the case, we need to stick close to it.

* It's not only a list of questions, it's a list of related questions and answers. The answer part (the snippet) is important.

Reply:
So, after confirmation on slack, the wording "SmartSnippet" is used throughout the admin UI when you configure this feature.

Which also means it will be reflected in the general documentation.

So we need to stick close.

Reply:
Ok, I guess if I go back to my initial impressions, the thing that caught my eye, aside from the plural, was the word "suggestions" being commoditized, like "value" or "result". I suggested "RelatedQuestions" since it's the key the api is using to return the values, even though there are answers stored in there too.

Mentioning `QuestionsAndAnswers` would be more accurate, but I don't think it's necessary 🤷 

Staying inline with the marketing, another option could be:
- `SmartSnippetQuestionsList`

Reply:
Renamed to `SmartSnippetQuestionsList`